Source/WebKit/PdfJSGResources.cmake

```diff
@@ -1,24 +1,32 @@
-file(GLOB_RECURSE PdfJSFiles CONFIGURE_DEPENDS ${THIRDPARTY_DIR}/pdfjs/*)
-
 set(PdfJSExtraFiles
     ${WEBCORE_DIR}/Modules/pdfjs-extras/content-script.js
 )
 
+# DEPFILE is not supported by Makefile generators until 3.20.
+if (${CMAKE_VERSION} VERSION_LESS "3.20" AND NOT ${CMAKE_GENERATOR} STREQUAL "Ninja")
+    message(FATAL_ERROR "Building PDF.js with Makefiles requires CMake 3.20 or newer. You should enable Ninja by passing -GNinja. Alternatively, either disable PDF.js with -DENABLE_PDFJS=OFF, or upgrade to CMake 3.20 or newer.")
+endif ()
+
 macro(WEBKIT_BUILD_PDFJS_GRESOURCES _derived_sources_dir)
+    # Remove these cmake_policy() calls after upgrading to cmake_minimum_required() 3.20.
+    cmake_policy(PUSH)
+    cmake_policy(SET CMP0116 OLD)
     add_custom_command(
         OUTPUT ${_derived_sources_dir}/PdfJSGResourceBundle.xml
         DEPENDS ${TOOLS_DIR}/glib/generate-pdfjs-gresource-manifest.py
-                ${PdfJSFiles}
         COMMAND ${PYTHON_EXECUTABLE} ${TOOLS_DIR}/glib/generate-pdfjs-gresource-manifest.py --input=${THIRDPARTY_DIR}/pdfjs --output=${_derived_sources_dir}/PdfJSGResourceBundle.xml
+        DEPFILE ${_derived_sources_dir}/PdfJSGResourceBundle.deps
```

Comment:
This is not right. This causes the manifest to depend on the *contents* of the files, but it really needs to depend on the *set* of files that will change.

Reply:
More complicated solution incoming...